Оптимизаций кода + изменение документации [WIP]

Rysik5318 opened the pull request:
## Описание
- Измение документации
- Оптимизация кода
- Редактирование summary в ClientSamgkOutputResponse


**Прочая информация**:

<br />

## Типы изменений
- [x] Исправление ошибки (не нарушающее нормальную работу изменение, устраняющее проблему)
- [x] Новая функция (не нарушающее функциональность изменение, добавляющее функциональность)
- [x] Сильное изменение (исправление или функция, которая приведет к изменению существующей функциональности)
- [x] Документация
<br />

## Контрольный список для принятия pull request
- [x] Я проверил, что проект может быть скомпилирован
- [x] Я протестировал свои изменения, и они сработали так, как ожидалось

### Другое
- [x] Требуется дополнительное тестирование

Rysik5318 committed:
Оптимизаций кода + изменение документации

Rysik5318 committed:
Микроизмения

TheCrazyWolf commented:
@maksim789456 глянь что здесь произошло

maksim789456 commented on ClientSamgk/Common/CommonSamgkController.cs:
```diff
@@ -16,25 +13,51 @@ namespace ClientSamgk.Common;
+        return TryDeserializeSafe<T>(restResponse.Content);
+    }
+
+    //protected async Task SendRequest(string url, Method method = Method.Get, object? body = null) // ����� �� ������������
```
Коментарий написан в win-1251 хотя все файлы в проекте в UTF-8

maksim789456 commented on ClientSamgk/Common/CommonSamgkController.cs:
```diff
@@ -46,78 +69,73 @@ public class CommonSamgkController : CommonCache
+        }
 
-        CabsCache = new List<LifeTimeMemory<IResultOutCab>>();
+        // ������� ���
```
Опять win-1251

maksim789456 commented on ClientSamgk/Controllers/ScheduleController.cs:
```diff
@@ -211,193 +226,348 @@ public async Task<IList<IResultOutScheduleFromDate>> GetScheduleAsync(DateOnly s
     }
 
+    // Возвращает соответствующий кеш в зависимости от типа поиска
+    IEnumerable<object> getCacheByType(ScheduleSearchType type) => type switch
```
Не хватает модификатора доступа, нейминг надо исправить до `GetCacheByType`

maksim789456 commented on ClientSamgk/Controllers/ScheduleController.cs:
```diff
@@ -211,193 +226,348 @@ public async Task<IList<IResultOutScheduleFromDate>> GetScheduleAsync(DateOnly s
+    };
+
+    // Возвращает идентификатор для элемента кеша
+    string getIdByType(ScheduleSearchType type, object item) => type switch
```
Такая же история

maksim789456 commented on ClientSamgk/Controllers/ScheduleController.cs:
```diff
@@ -211,193 +226,348 @@ public async Task<IList<IResultOutScheduleFromDate>> GetScheduleAsync(DateOnly s
+    string GetScheduleUrl(DateOnly date, ScheduleSearchType type, string id)
     {
-        return type switch
+        //return type switch // Хотя можно и так, но я люблю изобретать велосипед
```
Вместо коментирования старого кода, заменяй его полностью
В истории git'а он все равно останется

maksim789456 commented on ClientSamgk/Interfaces/Client/ICabController.cs:
```diff
@@ -5,57 +5,66 @@ namespace ClientSamgk.Interfaces.Client;
 {
     /// <summary>
-    /// Получить список кабинетов
+    /// Получить список кабинетов.
```
Точки в данном случае абслолютно тут не к чему

maksim789456 commented on ClientSamgk/Utils/HeadersUtils.cs:
```diff
@@ -6,16 +6,20 @@ public static class HeadersUtils
+        });
     }
+
+    //private static ICollection<KeyValuePair<string,string>> GetHeaders() // ����� ��� ��� ������ ��������� �������? (�������) P.s. � �����, ����� �������� dickPick
```
Опять какая то беда с кодировкой

maksim789456 commented on ClientSamgkOutputResponse/Implementation/Cabs/ResultOutCab.cs:
```diff
@@ -2,19 +2,19 @@
+    public string Auditory => getAuditory(Adress);
 
-    private string GetCampus(string fullAdress)
+    string getCampus(string fullAdress)
```
Очень странные изменения, к чему было убирать модификатор доступа и кривить нейминг 0_o

maksim789456 commented on ClientSamgkOutputResponse/Interfaces/Groups/IResultOutGroup.cs:
```diff
@@ -5,20 +5,22 @@ namespace ClientSamgkOutputResponse.Interfaces.Groups;
+
     /// <summary>
-    /// ID куратора группы, которые не работает и не совпадает с реальной жизнью
+    /// ID куратора группы, которые не работает и не совпадает с реальной жизнью. // Правда
```
Коментарий не в тему

maksim789456 commented on ClientSamgkOutputResponse/Implementation/Identity/ResultOutIdentity.cs:
```diff
@@ -6,11 +6,18 @@ public class ResultOutIdentity : IResultOutIdentity
+    //    var arraysTeacherName = Name.Replace("  ", string.Empty).Replace("  ", string.Empty).Split(' ');
+    //    return arraysTeacherName.Length >= 3 ? $"{arraysTeacherName[0]} {arraysTeacherName[1].FirstOrDefault()}. {arraysTeacherName[2].FirstOrDefault()}." : Name;
+    //}
+    string getShortName() // ���������
```
Кодировка, нарушение нейминга, отсуствия модификатора доступа

maksim789456 reviewed (changes requested):
В целом все хорошо, но есть моменты которые надо исправить:

- Кое где кривая кодировка коментариев, win-1251 или подобное. Все файлы в проекте в UTF-8
- Оставленный закоментированный старый код. Не стоит так делать т.к. git хранит историю изменений, так что его надо убрать
- В многих местах неправильное наименование методов, полей. Придерживайся гайдлайнов от [Microsoft](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members)
- Убирание модификаторов доступа. Мы все прерасно помним что сущности без модификатора доступа будут как private, но не надо это доводить до крайности и убирать намерено модификаторы. Код читается более удобнее если модификатор написан явно
- Неуместные коментарии по коду. Не надо превращять кодовую базу в свои личные заметки. Так же если ты хочешь что-то еще доделать, то создавай PR как draft

Rysik5318 committed:
Оптимизация ScheduleCallsExtensions

Оптимизация ScheduleCallsExtensions
Возвращение в HeadersUtils метод GetHeaders()

Rysik5318 committed:
Update ResultOutCab.cs

TheCrazyWolf reviewed (changes requested):
Доработать нейминг, кодировку

Rysik5318 committed:
Микроизменение

TheCrazyWolf committed:
Обновил кодировку

Rysik5318 committed:
Временное возвращение ScheduleCallsExtensions

maksim789456 committed:
Update encoding (again)

Rysik5318 committed:
Update ScheduleCallsExtensions.cs

Rysik5318 commented:
Вроде бы всё

Rysik5318 committed:
Update Program.cs

Rysik5318 committed:
Update ResultOutCab.cs

Rysik5318 committed:
Merge branch 'development' into master

Rysik5318 committed:
Добавление примеров + фикс кода Макса

Rysik5318 committed:
Update README.md

maksim789456 committed:
Cleanup

* Wrap long line (>130 chars)
* Remove unnecessary spaces, commas
* Fix spacing
* Fix multiple enumeration﻿

maksim789456 committed:
Docs for schedule get methods

maksim789456 committed:
Remove unnecessary funcs, inlining code

Calculate boolean values in function can be interfere compiler inlining

maksim789456 committed:
Ternary instead bulky if/else

Bulky if/else statements make it difficult to read code quickly

maksim789456 committed:
Add check at possible NPE place

maksim789456 reviewed (approved):
Все окей кроме нескольких моментов, но их сам поправлю

Rysik5318 committed:
Update README.md

TheCrazyWolf committed:
Убрал не нужную выборку

TheCrazyWolf committed:
Извлек получение нужного параметра по перечислению

TheCrazyWolf committed:
Исправил опечатку

TheCrazyWolf committed:
Set version to 3.0.1
